common/alerts.js

```diff
@@ -10,12 +10,15 @@
             if (alert.hasOwnProperty('type') && alert.hasOwnProperty('message')) {
                 return alerts.push(alert);
             }
-            console.log('Invalid alert properties: ', alert);
+            throw "Invalid alert";
         }
 
         function deleteAlert(alert) {
             var index = alerts.indexOf(alert);
-            alerts.splice(index, 1);
+            if (index > -1) {
+                return alerts.splice(index, 1);
+            }
+            throw "Alert not found.";
         }
 
         return {
```

Comment:
The following (3 lines)[https://github.com/informatics-isi-edu/chaise/blob/rec-errors/common/alerts.js#L25-L27] we should wrap each method with the `catchWrapper()`.